src/MovieDetails/MovieDetails.js

```diff
@@ -0,0 +1,204 @@
+      ratingId = this.state.currentUserRating.id
+      userId = this.props.currentUser.id
+
+      deleteUserRating(userId, ratingId)
+      .then(response => console.log('delete response', response))
+      .catch(error => this.setState({ error: error.message }))
+
+      this.createNewRating(userId, rating)
+
+    } else if (this.props.currentUser) {
+      userId = this.props.currentUser.id
+
+      this.createNewRating(userId, rating)
+    }
+  }
+
+  createNewRating = (userId, rating) => {
+    const newRating = {
+      movie_id: this.state.singleMovie.id,
+      rating: +rating
+    }
+    
+    console.log('newRating obj', newRating)
+    
+    postUserRating(userId, newRating)
+    .then(() => this.updateUserRating())
+    .catch(error => this.setState({ error: error.message }))
+  }
+
+  getUserRatings() {
+    this.setState({ formattedRating: this.state.singleMovie.average_rating.toFixed(1)})
+   
+    if (this.props.currentUser) {
+      this.updateUserRating()
+    }
+  }
+
+  updateUserRating = () => {
+    fetchUserRatings(this.props.currentUser.id)
+    .then(ratings => { 
+      console.log('allratings', ratings)
+      const userRating = ratings.ratings.find(rating => {
+        return rating.movie_id === this.state.singleMovie.id
+      })
+      console.log('userRating', userRating)
+      this.setState({ currentUserRating: userRating })
+    })
+    .catch(error => this.setState({ error: error.message}))  
+  }
+
+  formatGenres = () => {
+    if (!this.state.singleMovie.genres) {
+      return <p className="unavailable-genres">Unavailable</p>
+    } else if (this.state.singleMovie.genres.length > 1) {
+      let commaList = this.state.singleMovie.genres.map(genre => {
+        return(
+          <p key={Math.random()} className="genre">{genre}</p>
+        )
+      })
+      return commaList;
+    } else {
+      return <p className="genre">{this.state.singleMovie.genres[0]}</p>;
+    }
+  }
+
+  generateListItem() {
+    return (
+      <section>
+        <ListItem 
+          key={Math.random()}
+          label="Release Date"
+          body={new Date(this.state.singleMovie.release_date).toLocaleDateString()}
+        />
+        <ListItem 
+          key={Math.random()}
+          label="Runtime"
+          body={`${this.state.singleMovie.runtime} mins`}
+        />
+        <ListItem 
+          key={Math.random()}
+          label="Budget"
+          body={`$${new Intl.NumberFormat("en-US").format(this.state.singleMovie.budget)}`}
+        />
+        <ListItem 
+          key={Math.random()}
+          label="Revenue"
+          body={`$${new Intl.NumberFormat("en-US").format(this.state.singleMovie.revenue)}`}
+        />
+      </section>
+    )
+  }
+
+  toggleWatchlist = () => {
+    if (this.state.onWatchlist === false) {
+      this.setState({ onWatchlist: true })
+    } else {
+      this.setState({ onWatchlist: false})
+    }
+  }
+
+
+  render() {
+    return(
```

Comment:
This render is pretty big. Consider creating components from most of these sections.